examples/reminders/lib/feature_reminder_list/views/add_reminder_dialog.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:provider/provider.dart';
+
+import '../../base/models/reminder_model.dart';
+import '../../feature_reminder_manage/blocs/reminder_manage_bloc.dart';
+
+class AddReminderDialog extends StatefulWidget {
+  static int nextReminderId = 1000000;
+
+  const AddReminderDialog({Key? key}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => AddReminderDialogState();
+}
+
+class AddReminderDialogState extends State<AddReminderDialog> {
+  late TextEditingController _textEditingController;
+
+  late final FocusNode _titleFocusNode = FocusNode();
+
+  DateTime dueDate = DateTime.now();
+
+  final _formatter = DateFormat.yMd();
+
+  @override
+  void initState() {
+    _textEditingController = TextEditingController();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+        title: const Text('Add Reminder'),
+        content: Row(
+          children: [
+            Expanded(
+              child: TextField(
+                focusNode: _titleFocusNode,
+                textInputAction: TextInputAction.done,
+                controller: _textEditingController,
+                minLines: 1,
+                maxLines: 3,
+                decoration: const InputDecoration(
+                  border: UnderlineInputBorder(),
+                ),
+              ),
+            ),
+            TextButton(
+              onPressed: _onDueDatePressed,
+              child: Text(_formatter.format(dueDate)),
+            ),
+          ],
+        ),
+        actions: <Widget>[
+          TextButton(
+            child: const Text('OK'),
+            onPressed: () {
+              AddReminderDialog.nextReminderId++;
+              context.read<ReminderManageBlocType>().events.create(
+                  ReminderModel(
+                      id: AddReminderDialog.nextReminderId.toString(),
```

Comment:
Let's create the id in the repo/data source layer. The UI layer should not be responsible for creating such kind of ids.